legacy/src/main/java/com/amazon/opendistroforelasticsearch/sql/legacy/plugin/RestSQLQueryAction.java

```diff
@@ -109,6 +121,13 @@ public RestChannelConsumer prepareRequest(SQLQueryRequest request, NodeClient no
       return NOT_SUPPORTED_YET;
     }
 
+    Format format = request.format();
+    if (format.equals(Format.CSV)) {
+      formatter = new CsvResponseFormatter(request.escape());
+    } else {
+      formatter = new JdbcResponseFormatter(PRETTY);
+    }
```

Comment:
Ok, but in this case I extracted another property `request: SqlRequest` in `RestSQLQueryAction` to pass the sql request properties (including the format and sanitize option) to the listener

Reply:
I just realized this `formatter` is actually only required by `createQueryResponseListener` instead of both execute and explain. In this case, how about we just pass `request` and create specific formatter in `createQueryResponseListener`?

Reply:
Yes that's also what I'm thinking about, the changes are done